From 88ae787da56a3888de77b846063b34dbdf2f586c Mon Sep 17 00:00:00 2001
From: Lars Kanis <lars@greiz-reinsdorf.de>
Date: Fri, 4 Dec 2015 17:53:12 +0100
Subject: [PATCH 1/2] Fix cross build when srcdir is outside of builddir.

This brings the code in sync with template/fake.rb.in .
---
 tool/fake.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tool/fake.rb b/tool/fake.rb
index 9fe95a5..65c31ec 100644
--- a/tool/fake.rb
+++ b/tool/fake.rb
@@ -46,7 +46,7 @@ class File
   end
   join = proc {|*args| File.join(*args).sub!(/\A(?:\.\/)*/, '')}
   $topdir ||= builddir
-  $top_srcdir ||= join[$topdir, srcdir]
+  $top_srcdir ||= File.realpath(srcdir, $topdir)
   $extout = '$(topdir)/.ext'
   $extout_prefix = '$(extout)$(target_prefix)/'
   config = RbConfig::CONFIG
-- 
2.5.0

